src/main/resources/config/liquibase/changelog/20210920225828_changelog.xml

```diff
@@ -0,0 +1,103 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-3.9.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.9.xsd">
+    <changeSet author="argertboja" id="20210920225828-1">
+        <sql>
+            <!--This query finds all feedbacks which need a migration and prepares all the required
+            information to create all the missing text blocks such as text, first_index, end_index,
+            submission_id to which these text blocks belong to, type defined as MANUAL, the feedback_id
+            which is used to update the reference and the text block id which is computed as follows:
+            sha1(submission_id;start_index-end_index;text)
+            -->
+            CREATE TEMPORARY TABLE feedbacks_to_migrate
+            SELECT
+                f.reference AS text,
+                POSITION(f.reference IN s.text) AS start_index,
+                POSITION(f.reference IN s.text) + LENGTH(f.reference) AS end_index,
+                r.submission_id,
+                'MANUAL' AS TYPE,
+                f.id AS feedback_id,
+                SHA1(CONCAT(r.submission_id, ';', POSITION(f.reference IN s.text), '-', POSITION(f.reference IN s.text) + LENGTH(f.reference), ';', f.reference)) AS tb_id
```

Comment:
Just so that I understood it correctly this hashing here refers to the same concept we have in the code when we compute id at https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/domain/TextBlock.java#L88 
```
...
final String idString = submissionId + ";" + startIndex + "-" + endIndex + ";" + text;
id = sha1Hex(idString);
```
I was thinking about the difference between **startIndex** and the SQL's **POSITION(...)** function. I see that in our code **startIndex** starts from 0, but the SQL's [POSITION()](https://www.w3schools.com/mysql/func_mysql_position.asp) function starts from index 1 instead, and only returns 0 if nothing is found. Do you think this could be problematic or might create inconsistencies. What's your though on that?

Reply:
Yes, hashing refers to the same method we use to compute the text block id [here](https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/domain/TextBlock.java#L88).

We actually saw the POSITION() function together with @jpbernius however, we came to the conclusion that that shouldn't be a problem in this case since in the end we are getting the text correctly from submission text. Also, we seemed to find some cases where the text of text_block starts from 1 and not 0 in the existing text_blocks.  
However, in this case, we could also fix it by subtracting 1 from both start_index and end_index if  @jpbernius and @krusche think that would be better!

Reply:
If that was intended and worked well, then all fine by me 👍